Index: app/src/main/java/com/ifmg/carteiramensal/CadastroEdicaoEvento.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/ifmg/carteiramensal/CadastroEdicaoEvento.java	(revision 332f733fe22a5084f5461041f6fa6abfe7acc9e3)
+++ app/src/main/java/com/ifmg/carteiramensal/CadastroEdicaoEvento.java	(revision 326a6b88ba35e9960672be5d8de4a6946096416f)
@@ -6,18 +6,22 @@
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
+import android.widget.ArrayAdapter;
 import android.widget.Button;
 import android.widget.CheckBox;
 import android.widget.DatePicker;
 import android.widget.EditText;
 import android.widget.ImageView;
+import android.widget.Spinner;
 import android.widget.TextView;
 import android.widget.Toast;
 
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Calendar;
 import java.util.Date;
+import java.util.List;
 
 import ferramentas.EventosDB;
 import modelo.Evento;
@@ -35,6 +39,7 @@
     private Button salvarBtn;
     private Button cancelarBtn;
     private Calendar calendarioTemp;
+    private Spinner mesesRepeteSpi;
 
     //0 - cadastro de entrada, 1- cadastro de saída, 2- edição de entrada, 3- edição de saída
     private int acao = -1;
@@ -54,13 +59,26 @@
         fotoBtn = (Button) findViewById(R.id.fotoBtn);
         salvarBtn = (Button) findViewById(R.id.salvarCadastroBtn);
         cancelarBtn = (Button) findViewById(R.id.cancelarCadastroBtn);
+        mesesRepeteSpi = (Spinner) findViewById(R.id.mesesSpiner);
 
         Intent intencao = getIntent();
         acao = intencao.getIntExtra("acao", -1);
 
         ajustaPorAcao();
         cadastraEventos();
+        confSpinners();
 
+    }
+    private void confSpinners(){
+        List<String> meses = new ArrayList<>();
+        //vamos permitir nessa versão a repetição de apenas 24 meses de um evento
+        for(int i = 1; i<= 24; i++){
+            meses.add(i+"");
+        }
+        ArrayAdapter<String> listaAdapter = new ArrayAdapter<String>(this,
+                R.layout.support_simple_spinner_dropdown_item, meses);
+        mesesRepeteSpi.setAdapter(listaAdapter);
+        mesesRepeteSpi.setEnabled(false);
     }
 
     private void cadastraEventos(){
@@ -87,6 +105,26 @@
                 cadastrarNovoEvento();
             }
         });
+        //Tratando a repetição do evento
+
+        repeteBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if(repeteBtn.isChecked()){
+                    mesesRepeteSpi.setEnabled(true);
+                }else{
+                    mesesRepeteSpi.setEnabled(false);
+                }
+            }
+        });
+
+        cancelarBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //Termina a execuçao de uma Activity e retorna a anterior
+                finish();
+            }
+        });
     }
 
 
@@ -130,12 +168,12 @@
             valor *= -1;
         }
 
-        SimpleDateFormat formatador = new SimpleDateFormat("dd/MM/yyyy");
+        //SimpleDateFormat formatador = new SimpleDateFormat("dd/MM/yyyy");
+        //String dataStr = dataTxt.getText().toString();
 
-        String dataStr = dataTxt.getText().toString();
-        try {
+        //try {
 
-            Date diaEvento = formatador.parse(dataStr);
+            Date diaEvento = calendarioTemp.getTime();
             //um novo calendario para calcular a data limite (repetição)
             Calendar dataLimite = Calendar.getInstance();
             dataLimite.setTime(calendarioTemp.getTime());
@@ -143,6 +181,8 @@
             //verificando se esse evento irá repetir por alguns meses
             if(repeteBtn.isChecked()){
             //por enquanto estamos considerando apenas um mes
+                String mesStr = (String)mesesRepeteSpi.getSelectedItem();
+                dataLimite.add(Calendar.MONTH, Integer.parseInt(mesStr));
             }
             //setando para o ultimo dia do mes limite
             dataLimite.set(Calendar.DAY_OF_MONTH, dataLimite.getActualMaximum(Calendar.DAY_OF_MONTH));
@@ -157,9 +197,9 @@
 
             finish();
 
-        }catch (ParseException ex){
+        /*}catch (ParseException ex){
             System.err.println("erro no formato da data....");
-        }
+        }*/
 
     }
 
Index: app/src/main/java/com/ifmg/carteiramensal/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/ifmg/carteiramensal/MainActivity.java	(revision 332f733fe22a5084f5461041f6fa6abfe7acc9e3)
+++ app/src/main/java/com/ifmg/carteiramensal/MainActivity.java	(revision 326a6b88ba35e9960672be5d8de4a6946096416f)
@@ -10,9 +10,11 @@
 import android.widget.TextView;
 import android.widget.Toast;
 
+import java.util.ArrayList;
 import java.util.Calendar;
 
 import ferramentas.EventosDB;
+import modelo.Evento;
 
 public class MainActivity extends AppCompatActivity {
 
@@ -26,7 +28,7 @@
     private Button proxBtn;
     private Button novoBtn;
     private Calendar hoje;
-    private Calendar dataApp;
+    static Calendar dataApp;
 
 
     @Override
@@ -54,6 +56,7 @@
         hoje = Calendar.getInstance();
 
         mostraDataApp();
+        atualizaValores();
 
     }
     private void cadastroEventos(){
@@ -89,7 +92,7 @@
 
                 trocaAct.putExtra("acao", 0);
 
-                startActivity(trocaAct);
+                startActivityForResult(trocaAct, 0);
             }
         });
 
@@ -100,7 +103,7 @@
 
                 trocaAct.putExtra("acao", 1);
                 //pedimos para iniciar a activity passada como parametro
-                startActivity(trocaAct);
+                startActivityForResult(trocaAct, 1);
             }
         });
 
@@ -130,6 +133,35 @@
 
 
         mostraDataApp();
+        atualizaValores();
     }
+
+    private  void atualizaValores(){
+        //buscando as entradas e as saídas nesse memo banco de dados
+        EventosDB db = new EventosDB(MainActivity.this);
+
+        ArrayList<Evento> saidasListas = db.buscaEvento(1, dataApp);
+        ArrayList<Evento> entradasListas = db.buscaEvento(0, dataApp);
+
+        double entradaTotal = 0.0;
+        double saidaTotal = 0.0;
+
+        for(int i = 0; i< entradasListas.size(); i++){
+            entradaTotal += entradasListas.get(i).getValor();
+        }
+        for(int i = 0; i< saidasListas.size(); i++){
+            saidaTotal += saidasListas.get(i).getValor();
+        }
+        double saldoTotal = entradaTotal - saidaTotal;
 
+        entrada.setText(String.format("%2f", entradaTotal));
+        saida.setText(String.format("%2f", saidaTotal));
+        saldo.setText(String.format("%2f", saidaTotal));
+
+    }
+    protected void  onActivityResult(int codigoRequest, int codigoResultado, Intent data) {
+        super.onActivityResult(codigoRequest, codigoResultado, data);
+
+        atualizaValores();
+    }
 }
\ No newline at end of file
Index: app/src/main/java/com/ifmg/carteiramensal/VisualizarEventos.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/ifmg/carteiramensal/VisualizarEventos.java	(revision 332f733fe22a5084f5461041f6fa6abfe7acc9e3)
+++ app/src/main/java/com/ifmg/carteiramensal/VisualizarEventos.java	(revision 326a6b88ba35e9960672be5d8de4a6946096416f)
@@ -11,6 +11,11 @@
 import android.widget.TextView;
 import android.widget.Toast;
 
+import java.util.ArrayList;
+
+import ferramentas.EventosDB;
+import modelo.Evento;
+
 public class VisualizarEventos extends AppCompatActivity {
 
     private TextView tituloTxt;
@@ -19,6 +24,10 @@
     private Button novoBtn;
     private Button cancelarBtn;
 
+    private ArrayList<Evento> eventos;
+    private itemListaEvento adapter;
+
+
     //0peração = 0 indica entrada e operação = 1indica saída
     private int operacao = -1;
 
@@ -43,6 +52,8 @@
         cadastrarEventos();
         cadastrarEventos();
 
+        carregaEventosLista();
+
     }
 
     private void cadastrarEventos(){
@@ -55,15 +66,23 @@
                     Intent trocaAct = new Intent(VisualizarEventos.this, CadastroEdicaoEvento.class);
                     if(operacao == 0){
                         trocaAct.putExtra("acao",0);
+                        startActivityForResult(trocaAct, 0);
                     }else{
                         trocaAct.putExtra("acao", 1);
+                        startActivityForResult(trocaAct, 0);
                     }
 
-                    startActivity(trocaAct);
+
                 }
 
             }
         });
+         cancelarBtn.setOnClickListener(new View.OnClickListener() {
+             @Override
+             public void onClick(View v) {
+                 finish();
+             }
+         });
 
     }
 
@@ -84,7 +103,25 @@
 
     }
 
+    private void carregaEventosLista(){
+        EventosDB db = new EventosDB(VisualizarEventos.this);
+        eventos = db.buscaEvento(operacao, MainActivity.dataApp);
+
+        adapter = new itemListaEvento(getApplicationContext(), eventos);
+        listaEventos.setAdapter(adapter);
 
+        double total = 0.0;
+        for(int i = 0; i < eventos.size(); i++ ){
+            total += eventos.get(i).getValor();
+        }
+        totalTxt.setText(String.format("%2f", total));
+    }
+
+    protected void  onActivityResult(int codigoRequest, int codigoResultado, Intent data) {
+        super.onActivityResult(codigoRequest, codigoResultado, data);
+
+        carregaEventosLista();
+    }
 
 
 }
\ No newline at end of file
Index: app/src/main/java/ferramentas/EventosDB.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/ferramentas/EventosDB.java	(revision 332f733fe22a5084f5461041f6fa6abfe7acc9e3)
+++ app/src/main/java/ferramentas/EventosDB.java	(revision 326a6b88ba35e9960672be5d8de4a6946096416f)
@@ -2,10 +2,14 @@
 
 import android.content.ContentValues;
 import android.content.Context;
+import android.database.Cursor;
 import android.database.SQLException;
 import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteException;
 import android.database.sqlite.SQLiteOpenHelper;
 
+import java.util.ArrayList;
+import java.util.Calendar;
 import java.util.Date;
 
 import modelo.Evento;
@@ -50,8 +54,65 @@
     public void atualizaEvento(){
 
     }
-    public void buscaEvento(){
+    public ArrayList<Evento> buscaEvento(int op, Calendar data){
+
+        ArrayList<Evento> resultado = new ArrayList<>();
+        Calendar dia1 = Calendar.getInstance();
+        dia1.setTime(dia1.getTime());
+        dia1.set(Calendar.DAY_OF_MONTH,1);
+        dia1.set(Calendar.HOUR, -12);
+        dia1.set(Calendar.MINUTE, 0);
+        dia1.set(Calendar.SECOND, 0);
+        //ultimo dia do mes(ultimo minuto)
+        Calendar dia2 = Calendar.getInstance();
+        dia2.setTime(dia2.getTime());
+        dia2.set(Calendar.DAY_OF_MONTH, dia2.getActualMaximum(Calendar.DAY_OF_MONTH));
+        dia2.set(Calendar.HOUR, 11);
+        dia2.set(Calendar.MINUTE, 59);
+        dia2.set(Calendar.SECOND, 59);
+
+        String sql = "SELECT * FROM evento WHERE ((datavalida <=" + dia2.getTime().getTime() +
+                " AND dataovalida >=" + dia1.getTime().getTime()+") OR (dataocoreu <= "+ dia2.getTime().getTime()+
+                "AND datavalida>="+dia1.getTime().getTime()+"))";
+
+        sql+="AND valor ";
+
+        if(op == 0){
+            //Entradas
+            sql+= ">= 0";
+        }else{
+            //Saída(indicado por um valor negativo)
+            sql+="-< 0";
+        }
+
+        try(SQLiteDatabase db = this.getWritableDatabase()){
+            Cursor tuplas = db.rawQuery(sql, null);
+            //efetuar a leitura da tuplas
+            if(tuplas.moveToFirst()){
+                do{
+                    int id = tuplas.getInt(0);
+                    String nome = tuplas.getString(1);
+                    double valor = tuplas.getDouble(2);
+                    if(valor < 0 ){
+                        valor *=-1;
+                    }
+                    String urlfoto = tuplas.getString(3);
+                    Date dataocoreu = new Date(tuplas.getLong(4));
+                    Date datacadastro = new Date(tuplas.getLong(5));
+                    Date datavalida = new Date(tuplas.getLong(6));
+
 
+                    Evento temporario = new Evento((long)id, nome, valor, datacadastro, datavalida, dataocoreu, urlfoto);
+                    resultado.add(temporario);
+
+                }while(tuplas.moveToNext());
+            }
+
+        }catch (SQLiteException ex){
+
+        }
+
+        return resultado;
     }
 
 
Index: app/src/main/res/layout/activity_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main.xml	(revision 332f733fe22a5084f5461041f6fa6abfe7acc9e3)
+++ app/src/main/res/layout/activity_main.xml	(revision 326a6b88ba35e9960672be5d8de4a6946096416f)
@@ -33,11 +33,11 @@
             <!-- botão de cadastro de entrada-->
             <ImageButton
                 android:id="@+id/entradaBtn"
-                android:layout_width="70dp"
-                android:layout_height="70dp"
+                android:layout_width="50dp"
+                android:layout_height="50dp"
                 android:background="@mipmap/soma"
                 android:layout_gravity="center"
-                android:layout_marginRight="20dp"/>
+                android:layout_marginRight="10dp"/>
 
             <!--bloco lógico para organizar os TextView-->
             <LinearLayout
@@ -56,14 +56,14 @@
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
                         android:text="Entradas: R$"
-                        android:textSize="21dp"/>
+                        android:textSize="17dp"/>
 
                     <TextView
                         android:id="@+id/entradaTxt"
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
                         android:text="0,00"
-                        android:textSize="21sp"
+                        android:textSize="17sp"
                         android:layout_marginLeft="4dp"/>
 
                 </LinearLayout>
@@ -80,14 +80,14 @@
                         android:layout_width="wrap_content"
                         android:layout_height="match_parent"
                         android:text="Saídas: R$"
-                        android:textSize="21dp" />
+                        android:textSize="17dp" />
 
                     <TextView
                         android:id="@+id/saidaTxt"
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
                         android:text="0,00"
-                        android:textSize="21sp"
+                        android:textSize="17sp"
                         android:layout_marginLeft="4dp"/>/>
                 </LinearLayout>
 
@@ -102,14 +102,14 @@
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
                         android:text="saldo: R$"
-                        android:textSize="21dp"/>
+                        android:textSize="17dp"/>
 
                     <TextView
                         android:id="@+id/saldoTxt"
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
                         android:text="0,00"
-                        android:textSize="21sp"
+                        android:textSize="17sp"
                         android:layout_marginLeft="4dp"/>/>
 
                 </LinearLayout>
@@ -118,11 +118,11 @@
             <!-- botão de cadastro de saída-->
             <ImageButton
                 android:id="@+id/saidaBtn"
-                android:layout_width="70dp"
-                android:layout_height="70dp"
+                android:layout_width="50dp"
+                android:layout_height="50dp"
                 android:background="@mipmap/sub"
                 android:layout_gravity="center"
-                android:layout_marginLeft="20dp"/>
+                android:layout_marginLeft="10dp"/>
 
         </LinearLayout>
         <LinearLayout
Index: app/src/main/res/layout/activity_visualizar_eventos.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_visualizar_eventos.xml	(revision 332f733fe22a5084f5461041f6fa6abfe7acc9e3)
+++ app/src/main/res/layout/activity_visualizar_eventos.xml	(revision 326a6b88ba35e9960672be5d8de4a6946096416f)
@@ -18,10 +18,11 @@
             android:layout_height="wrap_content"
             android:text="Título"
             android:layout_gravity="center"
-            android:layout_marginTop="35dp"
+            android:layout_marginTop="15dp"
             android:layout_marginBottom="15dp"
             android:textSize="32sp"/>
 
+        <!-- Representando o total dos eventos da lista-->
         <ListView
             android:id="@+id/listaEventos"
             android:layout_width="match_parent"
@@ -29,9 +30,10 @@
             android:layout_marginLeft="15dp"
             android:layout_marginRight="15dp"
             android:layout_marginBottom="15dp"
-            android:layout_weight="1"/>
+            android:layout_weight="1"
+            android:padding="10dp"
+            android:background="@drawable/bordalista"/>
 
-        <!-- Representando o total dos eventos da lista-->
         <LinearLayout
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
